src/goto-instrument/contracts/contracts.cpp

```diff
@@ -387,10 +398,11 @@ void code_contractst::replace_old_parameter(
     replace_old_parameter(op, parameter2history, location, mode, history);
   }
 
-  if(expr.id() == ID_old)
+  if(expr.id() == ID_old ||
+     expr.id() == ID_loop_entry)
   {
     DATA_INVARIANT(
-      expr.operands().size() == 1, CPROVER_PREFIX "old must have one operand");
+      expr.operands().size() == 1, "history variable must have one operand");
```

Comment:
Why is this better than the previous message?

Reply:
I think Aalok was trying to generalize the error message for both `__CPROVER_old` and `__CPROVER_loop_entry`.

I just have a minor suggestion:
```suggestion
      expr.operands().size() == 1, "History variable must have one operand.");
```

Note that this is not a user-facing error message, but an internal invariant failure in case we tried to construct a bad history variable. But we can improve the invariant failure message, if this is unclear. Do you have any suggestions, @feliperodri?

Reply:
See https://github.com/diffblue/cbmc/pull/6278#discussion_r684582662

Reply:
I've made separate functions and avoided generalization.

Reply:
Please don't do that. That looks like a lot of duplicate code.

As I have commented below, this operand check can simply be removed.